src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6824,7 +6828,15 @@ final void TryProcessFeatureExtAck(boolean featureExtAckReceived) throws SQLServ
 final class TimeoutTimer implements Runnable {
     private final int timeoutSeconds;
     private final TDSCommand command;
-    private Thread timerThread;
+    private volatile Future<?> task;
+    private static final ExecutorService executor = Executors.newCachedThreadPool(new ThreadFactory() {
+        @Override
+        public Thread newThread(Runnable r) {
+            Thread t = Executors.defaultThreadFactory().newThread(r);
```

Comment:
Good idea. I will talk to the team and see what name we can give to the thread group. Thank you.